doc/logrotate/logrotate_hld.md

```diff
@@ -0,0 +1,239 @@
+
+# Scope
+The scope of this document is to cover definition, design and implementation of SONiC Log rotate configuration and log rotate CLI.
+The document covers the next CLI:
+1.	Commands to configure log rotate setting
+2.	Command to display log rotate settings
+
+<!-- omit in toc -->
+## Abbreviations
+| Term  | Meaning                                   |
+|:------|:------------------------------------------|
+| SONiC | Software for Open Networking in the Cloud |
+| DB    | Database                                  |
+| CLI   | Сommand-line Interface                    |
+| YANG  | Yet Another Next Generation               |
+
+<!-- omit in toc -->
+## List of figures
+- [Figure 1: Log rotate system chart diagram](#figure-1-logrotate-system-chart-diagram)
+- [Figure 2: Log rotate init flow](#figure-2-logrotate-init-flow)
+- [Figure 3: Log rotate config flow](#figure-3-logrotate-config-flow)
+- [Figure 4: Log rotate show configuration](#figure-4-logrotate-show-configuration)
+
+# 1 Introduction
+
+## 1.1 Feature overview
+This feature performs log rotation separately for syslog and debug files.
+The SONiC log rotate feature mantain several configuration parameter:
+|  Param            |  Description                                                 |
+| :---------------: | :----------------------------------------------------------: |
+|  disc-presentage  | Rotate logs when they surpass a specified percentage of disk |
+|  frequency        | Log files rotation frequency                                 |
+|  max-number       | Max number of log files to keep                              |
+|  size             | Rotate logs if they grow bigger then size in Mebibytes       |
+
+
+## 1.2 Requirements
+
+**This feature will support the following functionality:**
+1. Show log rotate configuration
+2. Configure log rotate
+    1. Disc-persentage
+    2. Frequency
+    3. Max-number
+    4. Size
+
+Configuration commands supports separate configuration for syslog and debug files.
+
+# 2 Design
+
+## 2.1 Overview
+Here is the representation of SONiC platform using logrotate feature:
+
+![Log rotate system chart diagram](images/logrotate_system_chart_diagram.png "Figure 1: Log rotate system chart diagram")
+<!-- omit in toc -->
+###### Figure 1: Log rotate system chart diagram
+
+This feature require access to SONiC DB. All log rotate related configuration (disc persentage, frequency and size, etc...) saved into the SONiC config database.
+Hostcfgd will listen for the configuration changes in corresponding tables and restart logrotate-config service. Log rotate config service - it is simple SystemD service.
+It reads log rotate configuration from database and apply it to log rotate config files.
+
+
+**The log rotate config files will be used:**
+1. /etc/logrotate.d/rsyslog - generic log files
+2. /etc/logrotate.d/debug - debug log files
+3. /etc/logrotate.d/syslog - syslog files
```

Comment:
Mentioning the contents of these files would be helpful.

Reply:
Additionally, while syslog is obvious, what files fall under debug log files? What about generic log files?

Reply:
Bump on this

Reply:
@fastiuk could you respond on this?